openwisp_notifications/static/openwisp-notifications/js/preferences.js

```diff
@@ -0,0 +1,445 @@
+                headers: { 'X-CSRFToken': $('input[name="csrfmiddlewaretoken"]').val() },
+                contentType: 'application/json',
+                data: JSON.stringify(data),
+                success: function () {
+                    showToast('success', gettext('Settings updated successfully.'));
+                },
+                error: function () {
+                    showToast('error', gettext('Something went wrong. Please try again.'));
+                    $(`.web-checkbox[data-organization-id="${organizationId}"][data-pk="${settingId}"]`).prop('checked', previousWebChecked);
+                    $(`.email-checkbox[data-organization-id="${organizationId}"][data-pk="${settingId}"]`).prop('checked', previousEmailChecked);
+                    updateOrgLevelCheckboxes(organizationId);
+                },
+                complete: function () {
+                    isUpdateInProgress = false;
+                }
+            });
+        });
+
+        $(document).on('change', '.main-checkbox', function () {
+            if (isUpdateInProgress) {
+                return;
+            }
+            const orgId = $(this).data('organization-id');
+            const triggeredBy = $(this).data('column');
+
+            let isOrgWebChecked = $(`.main-checkbox[data-organization-id="${orgId}"][data-column="web"]`).is(':checked');
+            let isOrgEmailChecked = $(`.main-checkbox[data-organization-id="${orgId}"][data-column="email"]`).is(':checked');
+
+            let previousOrgWebChecked, previousOrgEmailChecked;
+
+            if (triggeredBy === 'email') {
+                previousOrgEmailChecked = !isOrgEmailChecked;
+                previousOrgWebChecked = isOrgWebChecked;
+            } else {
+                previousOrgWebChecked = !isOrgWebChecked;
+                previousOrgEmailChecked = isOrgEmailChecked;
+            }
+
+            if (triggeredBy === 'email' && isOrgEmailChecked) {
+                isOrgWebChecked = true;
+            }
+
+            if (triggeredBy === 'web' && !isOrgWebChecked) {
+                isOrgEmailChecked = false;
+            }
+
+            isUpdateInProgress = true;
+
+            const data = {
+                web: isOrgWebChecked,
+                email: isOrgEmailChecked
+            };
+
+            $(`.main-checkbox[data-organization-id="${orgId}"][data-column="web"]`).prop('checked', isOrgWebChecked);
+            $(`.main-checkbox[data-organization-id="${orgId}"][data-column="email"]`).prop('checked', isOrgEmailChecked);
+
+            const table = $(this).closest('table');
+            table.find('.web-checkbox').prop('checked', isOrgWebChecked).change();
+            table.find('.email-checkbox').prop('checked', isOrgEmailChecked).change();
+
+            updateMainCheckboxes(table);
+
+            $.ajax({
+                type: 'POST',
+                url: `/api/v1/notifications/user/${userId}/organization/${orgId}/setting/`,
```

Comment:
Can you format the URL as we do it for the notification widget?, i.e prepend the `notificationApiHost`.

https://github.com/openwisp/openwisp-notifications/blob/63a29862e97faa1cc5847a149189be06171ee564/openwisp_notifications/static/openwisp-notifications/js/notifications.js#L316

Reply:
Updated.